Link to github: https://github.com/albert-moller/CSSAssignment1.git

Group members: Albert Frisch Møller (s214610) and Mark Andrawes (s214654)

For this assignment each group member contributed equally to every aspect of the assignment

### Assignment 1

#### Part 1: Web-Scraping

In [1]:
#Importing necessary Web Scraping and Data Storage packages
from bs4 import BeautifulSoup
import requests
import pandas as pd
import re
import os
from tqdm import tqdm
from joblib import Parallel, delayed
import networkx as nx
import json

if not os.path.exists("./data"):
    os.mkdir("./data")

In [2]:
LINK = "https://ic2s2-2023.org/program"
r = requests.get(LINK)
soup = BeautifulSoup(r.content)
researchers = []

table_rows = soup.find_all("tr")
for row in table_rows:
    for tag in row.find_all("a"):
        if re.search("Keynote", tag.text):
            string = tag.text
            string = string.replace("Keynote - ", "")
            string = string.strip()
            researchers.append(string)

plenaries = soup.find_all(class_='nav_list')

for plenary in plenaries:
    italic_names = plenary.find_all('i')
    italic_names = [i.get_text() for i in italic_names]

    for entry in italic_names:
        speakers = entry.split(',')
        for speaker in speakers:
            name = speaker.strip()
            if name not in researchers:
                researchers.append(speaker)

researchers_df = pd.DataFrame(researchers, columns=['Full Name'])
researchers_df.to_csv('data/ics2_researchers.csv', index=False)

print(f"The number of unique researchers is {len(researchers_df)}")

The number of unique researchers is 1856


#### How many unique researchers do you get?

We obtained that there are 1856 unique researchers present at the International Conference in Computational Social Science for 2023

##### Explain the process you followed to web-scrape the page. Which choices did you make to accurately retreive as many names as possible? Which strategies did you use to assess the quality of your final list? Explain your reasoning and your choices. (answer in max 150 words)

To webscrape the IC2S2 2023 program page, we inspected the HTML structure to identify the elements containing researcher names. Using BeautifulSoup, we collected the table rows ('tr') to extract the names of the keynote speakers. Additionally, we collected each 'i' tag within each plenary (found in the class 'nav_list') which contained multiple names of participants - we split these names and appended each one to the list. Finally, we stored the data in a Pandas DataFrame and saved as csv.


To retrieve the names accurately, we removed the 'Keynote -' prefix to cleanly retrieve the names of the keynote speakers. We used the '.strip()' function to remove any leading/trailing whitespace. This was done to ensure that we can accurately compare names when checking for duplicates. We assessed the quality by checking for duplicates by only adding new names to the list. This ensured the list only included unique participants.

